### PR TITLE
add parameter c

### LINKED ISSUE #5: Parameter -c

### PR DESCRIPTION
**Pull request by justinmueller05082000**
fixes #5

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -40,11 +47,16 @@ func main() {
...
 					break
 				}
+				count++
 				continue
-			} else if bytes.Compare(content,checkEquality) != 0 {
+			} else if bytes.Compare(content, checkEquality) != 0 {
 				if skip {
 					skip = false
+					if *countParameter {
+						fmt.Printf("      %v ", count)
```
Comment (Al2Klimov) [inline main.go]: Why v?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (justinmueller05082000) [reply main.go]: i dont know but it works

Comment (Al2Klimov) [reply main.go]: You don't know? Figure it out.

Comment (justinmueller05082000) [reply main.go]: v = the value in a default format

Comment (justinmueller05082000) [reply main.go]: but i should use %d i guess?

Comment (Al2Klimov) [reply main.go]: I didn't ask what it does, but why you chosen it.

Comment (Al2Klimov) [reply main.go]: Exactly.

File: main.go [hunk 2]
```diff
@@ -1,3 +1,5 @@
+// uniq | (c) 2020 NETWAYS GmbH | GPLv2+
+
```
Comment (Al2Klimov) [inline main.go]: Make another PR for this.

File: main.go [hunk 3]
```diff
@@ -13,11 +15,18 @@ func main() {
 	uniqueParameter := flag.Bool("u", false, "allows you to print only unique lines.")
 	repeatParameter := flag.Bool("d", false, "only print duplicate lines, one for each group.")
 	caseParameter := flag.Bool("i", false, "ignore differences in case when comparing.")
+	countParameter := flag.Bool("c", false, "prefix lines by the number of occurrences\n")
```
Comment (Al2Klimov) [inline main.go]: Why \n?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Why don't you check for EOF in all cases?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
There's almost always one line too many compared to the original.

**Comment by justinmueller05082000** [comment]
can you give me an example that shows exactly what you mean

**Comment by Al2Klimov** [comment]
```
uniq -c <<EOF
c
c
a
b
EOF
```

```
go run . -c <<EOF
c
c
a
b
EOF
```

**Comment by justinmueller05082000** [comment]
Screenshot from 2020-06-02 11-23-42

**Comment by justinmueller05082000** [comment]
Screenshot from 2020-06-02 11-25-24
is this the same problem as you mentioned above or is it another problem?

**Comment by Al2Klimov** [comment]
The second screenshot shows what I mean.

File: main.go [hunk 4]
```diff
@@ -40,11 +47,16 @@ func main() {
 				if readErr == io.EOF {
 					break
 				}
+				count++
 				continue
-			} else if bytes.Compare(content,checkEquality) != 0 {
+			} else if bytes.Compare(content, checkEquality) != 0 {
```
Comment (Al2Klimov) [inline main.go]: Not part of the task.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
* [x] `git reset --soft HEAD~$(git rev-list --count $(git merge-base HEAD master)..HEAD^)`
* [x] `git commit --amend --no-edit`
* [x] `git push -f origin feature/add-parameter-c-5`

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Write a test protocol.

**Comment by justinmueller05082000** [comment]
**Test protocol:**

**toUniq.txt:**

```
Dog
Cat
Mouse
Mouse
Mouse
Cat
Dog
Dog
Mouse
Mouse
```

**Go code:**

**Terminal:** `go run . -c < toUniq.txt `

**Output:**

```
      1 Dog
      1 Cat
      3 Mouse
      1 Cat
      2 Dog
      2 Mouse
```

**Uniq command:**

**Terminal:**  `uniq -c < toUniq.txt `

**Output:**

```
      1 Dog
      1 Cat
      3 Mouse
      1 Cat
      2 Dog
      2 Mouse
```


**Conclusion:**
Both results are equal.
